public/app/plugins/datasource/loki/datasource.test.ts

```diff
@@ -129,24 +114,14 @@ describe('LokiDatasource', () => {
   });
 
   describe('when doing logs queries with limits', () => {
-    const runTest = async (
-      queryMaxLines: number | undefined,
-      dsMaxLines: number | undefined,
-      expectedMaxLines: number
-    ) => {
-      let settings = {
-        url: 'myloggingurl',
+    const runTest = async (queryMaxLines: number | undefined, dsMaxLines: string, expectedMaxLines: number) => {
+      const settings = {
         jsonData: {
           maxLines: dsMaxLines,
         },
       } as DataSourceInstanceSettings<LokiOptions>;
 
-      const templateSrvMock = {
-        getAdhocFilters: (): any[] => [],
-        replace: (a: string) => a,
-      } as unknown as TemplateSrv;
-
-      const ds = new LokiDatasource(settings, templateSrvMock, timeSrvStub);
+      const ds = createLokiDatasource(templateSrvStub as unknown as TemplateSrv, settings);
```

Comment:
couldn't we do the ` as unknown as TemplateSrv` part where we define `templateSrvStub` (around line43) ? that way we only have to do it once and we don't have to do it when we use it.

Reply:
Absolutely. Good idea.